### DISCUSSION TITLE
loudness sampler messages

### DISCUSSION THREAD

**Discussion by alecristia** [LAAC-LSCP/ChildProject#282] [Q&A] [answered]
I sampled vanuatu with this code:

```
child-project sampler vanuatu samples/loud/ energy-detection \
  --windows-length 30000 --windows-spacing 300000 \
  --windows-offset 1800000 --windows-count 40 \
  --threshold .75 --low-freq 50 --high-freq 3000 \
  --by child_id	--profile standard
```

Messages look like this:
> computing the energy of 85 windows for recording child10/vanuatu_child10_20170812_1.wav...
computing the energy of 42 windows for recording child11/rec001.wav...
computing the energy of 42 windows for recording child11/rec002.wav...
computing the energy of 5 windows for recording child11/rec003.wav...
computing the energy of 2 windows for recording child11/rec004.wav...
computing the energy of 42 windows for recording child11/rec005.wav...
computing the energy of 42 windows for recording child11/rec006.wav...
computing the energy of 17 windows for recording child11/rec007.wav...
computing the energy of 174 windows for recording child12/vanuatu_child12_20170814_1.wav...
computing the energy of 0 windows for recording child12/vanuatu_child12_20170814_2.wav...
computing the energy of 119 windows for recording child12/vanuatu_child12_20170815_1.wav...
computing the energy of 42 windows for recording child13/rec001.wav...
computing the energy of 42 windows for recording child13/rec002.wav...
computing the energy of 42 windows for recording child13/rec003.wav...
computing the energy of 33 windows for recording child13/rec004.wav...

My Q is: how does the number of windows in this print out match up with the ones I asked? Is it perhaps that I gave inconsistent parameters? (that is, I intended to select 30-s windows, that are at least 300s apart, skipping the first 1800 s of the *recording*, and I aimed to have 40 such samples **per child** -- but perhaps the math doesn't work out the way I think when the --by-child parameter is used?)

**Comment by lucasgautheron** [comment, marked as answer]
The number printed out corresponds to the total amount of windows matching the length/spacing/offset specifications within each recording (=audio file). Signal energy is computed for each of these windows; only then the sampling process happens (at the requested level, i.e. child, session, or recording). For instance, there's a total of 192 windows for child11, and only 40 of those that pass the threshold ((1-0.75)*192=48) will be selected.

What this makes me realise is that, since many of those sessions are split across several recordings, it makes little sense to remove the first hour from all of the recordings within one session. So maybe we should change that... (in your case, what you really want is to remove the first hour from the first recording of each session)

**Reply by alecristia** [reply]
I think it makes sense to leave that parameter as is. It's true that session split due to tech reasons shouldn't have skips, but most of the time, splits happen because parents turn on/off the device, and since the skipping parameter is there to avoid sampling people who are hyper-aware, it's good to keep its behavior as it is.
Thanks for the super clear answer!
